akka-remote/src/main/scala/akka/remote/artery/tcp/ssl/SessionVerifier.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery.tcp.ssl
+
+import java.security.cert.X509Certificate
+
+import akka.annotation.ApiMayChange
+import javax.net.ssl.SSLSession
+
+/**
+ * Allows hooking in extra verification before finishing the SSL handshake.
+ */
+@ApiMayChange
+trait SessionVerifier {
```

Comment:
I can't see that any of these `SessionVerifier` are exposed in public API so we can make them internal?

Reply:
Good catch. This was public APi on a previous apprach I had. Privatizing...